src/esp/assets/ResourceManager.cpp

```diff
@@ -1577,56 +1568,55 @@ gfx::PbrMaterialData::uptr ResourceManager::buildPbrShadedMaterialData(
-  // the PbrShader (see PbrMaterialData for more details)
-
   // roughness
-  if (material.hasAttribute(Mn::Trade::MaterialAttribute::Roughness)) {
-    finalMaterial->roughness = material.roughness();
-  }
+  finalMaterial->roughness = material.roughness();
   if (material.hasRoughnessTexture()) {
     finalMaterial->roughnessTexture =
         textures_.at(textureBaseIndex + material.roughnessTexture()).get();
-    if (!material.hasAttribute(Mn::Trade::MaterialAttribute::Roughness)) {
-      finalMaterial->roughness = 1.0f;
-    }
   }
 
-  // metalness
-  if (material.hasAttribute(Mn::Trade::MaterialAttribute::Metalness)) {
-    finalMaterial->metallic = material.metalness();
-  }
+  // metallic
+  finalMaterial->metallic = material.metalness();
   if (material.hasMetalnessTexture()) {
     finalMaterial->metallicTexture =
         textures_.at(textureBaseIndex + material.metalnessTexture()).get();
-    if (!material.hasAttribute(Mn::Trade::MaterialAttribute::Metalness)) {
-      finalMaterial->metallic = 1.0f;
-    }
   }
 
-  // packed textures
-  if (material.hasOcclusionRoughnessMetallicTexture()) {
-    // occlusionTexture, roughnessTexture, metalnessTexture are pointing to the
-    // same texture ID, so just occlusionTexture
-    finalMaterial->occlusionRoughnessMetallicTexture =
-        textures_.at(textureBaseIndex + material.occlusionTexture()).get();
-  }
+  // sanity check when both metallic and roughness materials are presented
+  if (material.hasMetalnessTexture() && material.hasRoughnessTexture()) {
+    /*
+       sanity check using hasNoneRoughnessMetallicTexture() to ensure that:
+        - both use the same texture coordinate attribute,
+        - both have the same texture transformation, and
+        - the metalness is in B and roughness is in G
 
-  if (material.hasNoneRoughnessMetallicTexture()) {
-    // roughnessTexture, metalnessTexture are pointing to the
-    // same texture ID, so just roughnessTexture
-    finalMaterial->noneRoughnessMetallicTexture =
-        textures_.at(textureBaseIndex + material.roughnessTexture()).get();
-  }
+       It checks for a subset of hasOcclusionRoughnessMetallicTexture(),
+       so hasOcclusionRoughnessMetallicTexture() is not needed here.
 
-  if (material.isDoubleSided()) {
-    finalMaterial->doubleSided = true;
+       The normal/roughness/metallic is a totally different packing (in BA
+       instead of GB), and it is NOT supported in the current version.
+       so hasNormalRoughnessMetallicTexture() is not needed here.
+
+    */
+    CORRADE_ASSERT(
+        material.hasNoneRoughnessMetallicTexture(),
+        "ResourceManager::buildPbrShadedMaterialData(): if both the metallic "
+        "and roughness texture exist, they must be packed in the same texture "
+        "based on glTF 2.0 Spec.",
+        finalMaterial);
   }
 
+  // TODO:
+  // Support NormalRoughnessMetallicTexture packing
+  CORRADE_ASSERT(!material.hasNormalRoughnessMetallicTexture(),
```

Comment:
@mosra : FYI: I just added a corrade assertion here after you approved this diff.

Reply:
I just saw :) `hasNormalRoughnessMetallicTexture()` and `hasNoneRoughnessMetallicTexture()` are mutually exclusive, never both `true`, so *technically* this assert is unnecessary, however one extra check won't hurt I think.

Reply:
Oh, OK. Makes sense.

I would recommend you to put this sentence:
>hasNormalRoughnessMetallicTexture() and hasNoneRoughnessMetallicTexture() are mutually exclusive, never both true

to your wiki even it becomes so obvious after thinking about it for another second.